### ISSUE TITLE
NoSuchMethodException in Android Studio 2.2

### ISSUE THREAD

**Issue by Pitel** [pbreault/adb-idea#49]
```
java.lang.NoSuchMethodException: No similar method canRunOnDevice with params [class com.android.sdklib.AndroidVersion, class com.android.sdklib.repository.targets.PlatformTarget, class java.util.RegularEnumSet, class com.android.ddmlib.Device, class org.joor.Reflect$NULL] could be found on type class com.android.tools.idea.run.LaunchCompatibility.
org.joor.ReflectException: java.lang.NoSuchMethodException: No similar method canRunOnDevice with params [class com.android.sdklib.AndroidVersion, class com.android.sdklib.repository.targets.PlatformTarget, class java.util.RegularEnumSet, class com.android.ddmlib.Device, class org.joor.Reflect$NULL] could be found on type class com.android.tools.idea.run.LaunchCompatibility.
	at org.joor.Reflect.call(Reflect.java:406)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompatBefore2_0.getPreviousImplementation(CanRunOnDeviceCompatBefore2_0.java:37)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompatBefore2_0.getPreviousImplementation(CanRunOnDeviceCompatBefore2_0.java:13)
	at com.developerphil.adbidea.compatibility.BackwardCompatibleGetter.get(BackwardCompatibleGetter.java:17)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompat.getPreviousImplementation(CanRunOnDeviceCompat.java:36)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompat.getPreviousImplementation(CanRunOnDeviceCompat.java:12)
	at com.developerphil.adbidea.compatibility.BackwardCompatibleGetter.get(BackwardCompatibleGetter.java:14)
	at com.developerphil.adbidea.ui.MyDeviceChooser$MyDeviceTableModel.getValueAt(MyDeviceChooser.java:422)
	at javax.swing.JTable.getValueAt(JTable.java:2717)
	at com.intellij.ui.table.JBTable.calculateRowHeight(JBTable.java:178)
	at com.intellij.ui.table.JBTable.getRowHeight(JBTable.java:157)
	at javax.swing.JTable.getRowHeight(JTable.java:992)
	at javax.swing.JTable.getCellRect(JTable.java:2937)
	at javax.swing.JTable.valueChanged(JTable.java:4698)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:184)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:164)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:211)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:405)
	at javax.swing.DefaultListSelectionModel.changeSelection(DefaultListSelectionModel.java:415)
	at javax.swing.DefaultListSelectionModel.setSelectionInterval(DefaultListSelectionModel.java:459)
	at com.developerphil.adbidea.ui.MyDeviceChooser.refreshTable(MyDeviceChooser.java:296)
	at com.developerphil.adbidea.ui.MyDeviceChooser.access$600(MyDeviceChooser.java:68)
	at com.developerphil.adbidea.ui.MyDeviceChooser$6.run(MyDeviceChooser.java:275)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:345)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:329)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
	at java.awt.Dialog.show(Dialog.java:1084)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:792)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:465)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1661)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1610)
	at com.developerphil.adbidea.adb.AdbFacade.askUserForDevice(AdbFacade.java:120)
	at com.developerphil.adbidea.adb.AdbFacade.getDevice(AdbFacade.java:91)
	at com.developerphil.adbidea.adb.AdbFacade.executeOnDevice(AdbFacade.java:51)
	at com.developerphil.adbidea.adb.AdbFacade.restartDefaultActivity(AdbFacade.java:39)
	at com.developerphil.adbidea.action.RestartAction.actionPerformed(RestartAction.java:10)
	at com.developerphil.adbidea.action.AdbAction.actionPerformed(AdbAction.java:13)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:191)
	at com.intellij.ide.actions.GotoActionAction$7.run(GotoActionAction.java:275)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:345)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:329)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.NoSuchMethodException: No similar method canRunOnDevice with params [class com.android.sdklib.AndroidVersion, class com.android.sdklib.repository.targets.PlatformTarget, class java.util.RegularEnumSet, class com.android.ddmlib.Device, class org.joor.Reflect$NULL] could be found on type class com.android.tools.idea.run.LaunchCompatibility.
	at org.joor.Reflect.similarMethod(Reflect.java:473)
	at org.joor.Reflect.call(Reflect.java:403)
	... 75 more



org.joor.ReflectException: java.lang.NoSuchMethodException: No similar method canRunOnDevice with params [class com.android.sdklib.AndroidVersion, class com.android.sdklib.repository.targets.PlatformTarget, class java.util.RegularEnumSet, class com.android.ddmlib.Device, class org.joor.Reflect$NULL] could be found on type class com.android.tools.idea.run.LaunchCompatibility.
	at org.joor.Reflect.call(Reflect.java:406)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompatBefore2_0.getPreviousImplementation(CanRunOnDeviceCompatBefore2_0.java:37)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompatBefore2_0.getPreviousImplementation(CanRunOnDeviceCompatBefore2_0.java:13)
	at com.developerphil.adbidea.compatibility.BackwardCompatibleGetter.get(BackwardCompatibleGetter.java:17)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompat.getPreviousImplementation(CanRunOnDeviceCompat.java:36)
	at com.developerphil.adbidea.compatibility.CanRunOnDeviceCompat.getPreviousImplementation(CanRunOnDeviceCompat.java:12)
	at com.developerphil.adbidea.compatibility.BackwardCompatibleGetter.get(BackwardCompatibleGetter.java:14)
	at com.developerphil.adbidea.ui.MyDeviceChooser$MyDeviceTableModel.getValueAt(MyDeviceChooser.java:422)
	at javax.swing.JTable.getValueAt(JTable.java:2717)
	at com.intellij.ui.table.JBTable.calculateRowHeight(JBTable.java:178)
	at com.intellij.ui.table.JBTable.getRowHeight(JBTable.java:157)
	at javax.swing.JTable.rowAtPoint(JTable.java:2869)
	at javax.swing.plaf.basic.BasicTableUI.paint(BasicTableUI.java:1781)
	at javax.swing.plaf.ComponentUI.update(ComponentUI.java:161)
	at javax.swing.JComponent.paintComponent(JComponent.java:780)
	at com.intellij.ui.table.JBTable.paintComponent(JBTable.java:301)
	at javax.swing.JComponent.paint(JComponent.java:1056)
	at com.intellij.ui.table.JBTable.paint(JBTable.java:395)
	at javax.swing.JComponent.paintChildren(JComponent.java:889)
	at javax.swing.JComponent.paint(JComponent.java:1065)
	at javax.swing.JViewport.paint(JViewport.java:728)
	at com.intellij.ui.components.JBViewport.paint(JBViewport.java:165)
	at javax.swing.JComponent.paintChildren(JComponent.java:889)
	at javax.swing.JComponent.paint(JComponent.java:1065)
	at javax.swing.JComponent.paintToOffscreen(JComponent.java:5210)
	at javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1579)
	at javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1502)
	at javax.swing.RepaintManager.paint(RepaintManager.java:1272)
	at javax.swing.JComponent._paintImmediately(JComponent.java:5158)
	at javax.swing.JComponent.paintImmediately(JComponent.java:4969)
	at javax.swing.RepaintManager$4.run(RepaintManager.java:831)
	at javax.swing.RepaintManager$4.run(RepaintManager.java:814)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:814)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:789)
	at javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:738)
	at javax.swing.RepaintManager.access$1200(RepaintManager.java:64)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1732)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
	at java.awt.Dialog.show(Dialog.java:1084)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:792)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:465)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1661)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1610)
	at com.developerphil.adbidea.adb.AdbFacade.askUserForDevice(AdbFacade.java:120)
	at com.developerphil.adbidea.adb.AdbFacade.getDevice(AdbFacade.java:91)
	at com.developerphil.adbidea.adb.AdbFacade.executeOnDevice(AdbFacade.java:51)
	at com.developerphil.adbidea.adb.AdbFacade.restartDefaultActivity(AdbFacade.java:39)
	at com.developerphil.adbidea.action.RestartAction.actionPerformed(RestartAction.java:10)
	at com.developerphil.adbidea.action.AdbAction.actionPerformed(AdbAction.java:13)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:191)
	at com.intellij.ide.actions.GotoActionAction$7.run(GotoActionAction.java:275)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:345)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:329)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.NoSuchMethodException: No similar method canRunOnDevice with params [class com.android.sdklib.AndroidVersion, class com.android.sdklib.repository.targets.PlatformTarget, class java.util.RegularEnumSet, class com.android.ddmlib.Device, class org.joor.Reflect$NULL] could be found on type class com.android.tools.idea.run.LaunchCompatibility.
	at org.joor.Reflect.similarMethod(Reflect.java:473)
	at org.joor.Reflect.call(Reflect.java:403)
	... 89 more
```

**Comment by AHarazim** [comment]
Same Problem here with AS 2.2 Preview 4.

**Comment by henriquedesousa** [comment]
Same, AS 2.2 Preview 5

**Comment by pbreault** [comment]
Long overdue but fixed in 1.2.8
